### PR TITLE
Add setting to disable automatic fetching of subscription feed

### LINKED ISSUE #2515: [Feature Request]: Optionally disable automatic subscription fetching
### Guidelines

- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a feature request that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.

### Problem Description

For users with slow connections / many subscriptions, the behavior of automatically refreshing the subscriptions feed without input from the user can be frustrating and undesirable, as this process can take quite a long time and prevents usage of the app.  This is doubly true because a broad variety of seemingly unrelated behaviors can trigger an auto-refresh to occur, and it cannot be cancelled after it has begun.

### Proposed Solution

I seem to recall an option to refresh the feed exclusively manually in a previous version of the application, but I cannot locate it in the current build.  Adding an option to disable the auto-refresh behavior would greatly improve my experience as a user.

### Alternatives Considered

Increasing the speed of subscription fetching without encountering rate limiting is arguably the "correct" solution, but it's a much larger task, one that this project has tried to tackle before.  A simple toggle to only refresh the feed upon the user's request seems a suitable workaround for people with my usecase.

### Issue Labels

ease of use improvement, new optional setting

### Additional Information

v0.17.1 beta installed via AUR

### PR DESCRIPTION
**Pull request by Aiz0**
# Add setting to disable automatic fetching of subscription feed

## Pull Request Type

- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue

closes #2515
part of #539

### Main changes
Adds a new setting in subscription settings to disable automatic fetching of the subscription feed.
Prevents calls to the method `getSubscriptions()` in` Subscriptions.js` in to places.

1. When the subscriptions component is mounted
2. In `getProfileSubscriptions` method, when the users switches profile.

When the users switches profile and has disabled auto fetching 2 things can happen
1. the video list is emptied.
2. if the video list is saved in `profileSubscriptions` then it will be loaded instead.

### other stuff

In `Subscriptions.vue` I made some changes to the messages that show when the video list is empty
1. The regular "You have not subscribed to any channel" message will now only show when you have in fact not subscribed to any channel.
2. Added a new message that only shows if you have disabled automatic fetching.
3. Added a new message that will show when the user is subscribed to channels that have not uploaded any videos.

~~message 2 will block message 3 from being displayed even if they have attempted two fetch subscriptions.
This will probably not be a huge deal since I imagine it rarely happens.
The alternatives is to check if the user has attempted to refresh subscriptions.~~

## Screenshots 

This is the message that shows when the setting is disabled and the user starts the app or navigates to a new profile.
2022-09-28_T02:27:41

## Testing 

I created 2 new profiles
one without any channels to test if message 1 is displayed
the second one included only a channel without any videos to test if message 3 is displayed.

with the new setting disabled
I tested to make sure subscriptions where not fetched when freetube is started.
I switched between profiles to make sure subscriptions where not fetched automatically
I fetched subscriptions from one profile switched to another and the switched back to see if the video list was kept.

chosen api or rss should not affect anything since this check is done before that part of the code.

## Desktop

- **OS:** Arch Linux
- **OS Version:** 5.19.8-arch1-1
- **FreeTube version:** 0.17.1

## Additional context

It's likely that I've missed something or implemented this weirdly since I am unfamiliar with this part of the code.

I had to remove the newline for the `Your Subscription list is currently empty` string in the locale file to get the new strings to work.

If disabling fetching when switching profile is unwanted I could make a new setting for that or just remove it.

### CODE REVIEW

**Comment by Aiz0** [comment]
Accidentally opened this before filling in the template... It's filled in now so I'll reopen it.

**Review by PikachuEXE — COMMENTED** [review]
Code changes look fine
Still need to test locally tomorrow (21:08 here)

**Review by PikachuEXE — APPROVED** [review]
Tested
- Main changes
- Profile with no channel for message 1
- Switching between profiles to see fetched video list kept

message 3 not tested since no channel with no video provided

File: src/renderer/components/subscription-settings/subscription-settings.vue [hunk 1]
```diff
@@ -14,6 +14,12 @@
         :tooltip="$t('Tooltips.Subscription Settings.Fetch Feeds from RSS')"
         @change="updateUseRssFeeds"
       />
+      <ft-toggle-switch
```
Comment (ChunkyProgrammer) [inline src/renderer/components/subscription-settings/subscription-settings.vue:17]: I think you should add a switch column grid and two switch columns so it matches the layout of other setting components (parental control settings, distraction free settings, privacy settings, player settings)

**Comment by Aiz0** [comment]
> message 3 not tested since no channel with no video provided

Sorry forgot to include a one. Didn't want to pick a random users channel.
Youtube music channel works though since it's a special channel without any videos uploaded.
https://www.youtube.com/channel/UC-9-kyTW8ZkZNDHQJ6FgpwQ

File: src/renderer/components/subscription-settings/subscription-settings.vue (hunk 1 above)
Comment (Aiz0) [reply src/renderer/components/subscription-settings/subscription-settings.vue]: I've added one now, It does look nicer. 
2022-09-29_T18:49:07
Should I change their order?

File: src/renderer/components/subscription-settings/subscription-settings.vue [hunk 2]
```diff
@@ -2,25 +2,29 @@
   <ft-settings-section
     :title="$t('Settings.Subscription Settings.Subscription Settings')"
   >
-    <ft-flex-box class="settingsFlexStart500px">
```
Comment (absidue) [inline src/renderer/components/subscription-settings/subscription-settings.vue:5]: Can you please remove the ft-flex-box component and import from the `.js` file as well please.

Comment (Aiz0) [reply src/renderer/components/subscription-settings/subscription-settings.vue:5]: should be removed now unless i missed something

**Comment by Aiz0** [comment]
I was over complicating this in my mind...
The correct message should now display in all cases.

2022-09-29_T19:15:14

**Review by absidue — APPROVED** [review]

**Review by ChunkyProgrammer — DISMISSED** [review]

File: static/locales/en-US.yaml [hunk 3]
```diff
@@ -768,6 +770,8 @@ Tooltips:
     Fetch Feeds from RSS: When enabled, FreeTube will use RSS instead of its default
       method for grabbing your subscription feed. RSS is faster and prevents IP blocking,
       but doesn't provide certain information like video duration or live status
+    Fetch Automatically: When enabled, FreeTube will automatically fetch
+      your subscription feed on start and when switching profile.
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline static/locales/en-US.yaml:774]: Not sure if important to mention but feed will also be fetched when opening a new window/tab

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — COMMENTED** [review]
Some observations

File: static/locales/en-US.yaml (hunk 3 above)
Comment (absidue) [reply static/locales/en-US.yaml:774]: The visual space between them can be fixed by adding the `:compact="true"` attribute to all 3 switches.

Comment (Aiz0) [reply static/locales/en-US.yaml:774]: > Not sure if important to mention but feed will also be fetched when opening a new window/tab

I don't think _"on start"_ is the best wording.
I'll see if I can think of something better or if there's any suggestions.
Maybe _"when a window is opened"_ or something.

Comment (Aiz0) [reply static/locales/en-US.yaml:774]: > The visual space between them can be fixed by adding the `:compact="true"` attribute to all 3 switches.

Thanks, I'll add that.

Comment (Aiz0) [reply static/locales/en-US.yaml]: I've made the switches compact and reworded the tooltip.

**Review by PikachuEXE — APPROVED** [review]
All tested locally

**Review by ChunkyProgrammer — APPROVED** [review]

**Pull request merged**